backend/main_test.go

```diff
@@ -526,20 +531,41 @@ func (AuthClientStub) VerifyIDToken(ctx context.Context, idToken string) (*fireb
 }
 
 func TestFirebaseAuth(t *testing.T) {
-	auth := auth.FirebaseAuth{FirebaseClient: AuthClientStub{}}
+	authDomain := "google.com"
+	auth := auth.FirebaseAuth{
+		FirebaseClient: AuthClientStub{},
+		AuthDomain:     authDomain,
+	}
 	server := Server{store: in_memory_storage.MakeInMemStore(), auth: auth}
 	handler := server.makeHandler()
 
 	req := httptest.NewRequest(http.MethodGet, "/api/team/", nil)
-	req.Header.Add("Authentication", "Bearer pass")
+	req.Header.Add("Authorization", "Bearer pass")
 	resp := makeHTTPRequest(req, handler, t)
 	checkResponseStatus(http.StatusOK, resp, t)
 
-	req.Header.Set("Authentication", "Bearer passEmailUnverified")
+	req.Header.Set("Authorization", "Bearer passEmailUnverified")
 	resp = makeHTTPRequest(req, handler, t)
-	checkResponseStatus(http.StatusOK, resp, t)
+	checkResponseStatus(http.StatusUnauthorized, resp, t)
+
+	req.Header.Set("Authorization", "Bearer reject")
+	resp = makeHTTPRequest(req, handler, t)
+	checkResponseStatus(http.StatusUnauthorized, resp, t)
+
+	// Testing with unauthorised user domain
+	auth.AuthDomain = ""
```

Comment:
Can you not just say `server.auth.AuthDomain = ""`? I don't understand why you'd have to create a new `Server` here.